src/range.cr

```diff
@@ -252,6 +252,56 @@ struct Range(B, E)
     {% end %}
   end
 
+  # Returns the union of this range, and another.
+  # Returns 0..0 if there is no overlap.
+  #
+  # ```
+  # (1..5).union(5..10)   # => 1..10
+  # (1...10).union(1..10) # => 1..10
+  # ```
+  def union(other : Range)
+    if self.end < other.begin || other.end < self.begin
+      return 0..0
```

Comment:
I'm not sure we can get away with hardcoding a `Range(Int32, Int32)` specifically. My own use case for this functionality is actually `Range(Time, Time)`, so I'd get compile-time errors with this return value.

Reply:
Good catch, and thank you for the quick review! 

I'll get on these changes asap! ✌️⭐

Reply:
Using any of the four endpoints here instead of `0` should probably work too

Reply:
To me nil seems like a more logical return value than using any of the existing endpoints which might be confusing. 

Totally open to your reasoning though!

Reply:
Think the intersection of a list returning a nil vs an empty list: which feels more logical? Which is less of a burden?

Returning `nil` means that the methods _always_ return a nilable type (`Range(B, E) | Nil`) and we must _always_ check that whether the value is nil or not.

Returning an empty range means we always return a Range(B, E), which would act as a NOOP object.

Reply:
BTW: `0..0` wasn't an empty range, it's a range with a single value (starts at 0 and terminates at 0 inclusive):

```crystal
(0..0).to_a # => [0]
(0...0).to_a # => []
```